u_tls_extensions.go

```diff
@@ -54,10 +54,7 @@ func (e *SNIExtension) writeToUConn(uc *UConn) error {
 }
 
 func (e *SNIExtension) Len() int {
-	if len(e.ServerName) > 0 {
-		return 4 + 2 + 1 + 2 + len(e.ServerName)
-	}
-	return 0
```

Comment:
This caused the check at line 61 to stop functioning, because `len(b) < 0` is always false, so instead of returning `ErrShortBuffer` the code panics somewhere.